PocketKit/Sources/PocketKit/MyList/SearchItemsList/SearchItem.swift

```diff
@@ -40,8 +50,57 @@ struct SearchItem {
         ItemAction.share { _ in Log.info("Share button tapped!") }
     }
 
-    var favoriteAction: ItemAction {
-        ItemAction.favorite { _ in Log.info("Favorite button tapped!") }
+    func favoriteAction(index: Int, scope: SearchScope) -> ItemAction {
+        if isFavorite {
+            return .unfavorite { [weak self] _ in
+                guard let self else { return }
```

Comment:
@cyndichin when will this be nil? Can we sentry log it, this feels like a case we shouldn't ever want to hit

Reply:
This is due to weakifying self and it's a bit of a boiler plate code in Swift when handling this situation. If self is released, we don't want the closure to prevent self from being removed so we make it a weak self. If self is nil, then we don't want to trigger the code. I think it is nil when ARC frees up the memory and deallocates the instance. By doing the `guard` statement we are ensuring that we want a strong reference (instance exists) and so the object is available until execution of closure is complete at which point self can then be released. 

Not opposed to adding a sentry log, but maybe makes more sense as an `info` or `debug` message?

cc: @timc-mozilla @Gio2018 let me know if you have any additional input on this

Reply:
I decided to use optional self instead as it's better readability but please let me know if you would prefer the other method. I have no strong opinions on either.

Reply:
I agree @cyndichin . If we don't do this, we might end up with a memory leak. Another option if we don't want to do the unwrap, would be to use `unowned`, but only if we are 100% certain that `self` will never be nil when we call that closure, otherwise the app will crash.